jwst/lib/engdb_mast.py

```diff
@@ -91,6 +94,72 @@ def __init__(self, base_url=None, token=None, **service_kwargs):
                                      headers={'Authorization': f'token {token}'})
         self._session = requests.Session()
 
+    def cache(self, mnemonics, starttime, endtime, cache_path):
+        """Cache results for the list of mnemonics
+
+        Parameters
+        ----------
+        mnemonics: iterable
+            List of mnemonics to retrieve
+
+        starttime: str or astropy.time.Time
+            The, inclusive, start time to retireve from.
+
+        endttime: str or astropy.time.Time
+            The, inclusive, end time to retireve from.
+
+        cache_path: str or Path-like
+        Path of the cache directory.
+        """
+        cache_path = Path(cache_path)
+        cache_path.mkdir(parents=True, exist_ok=True)
+
+        for mnemonic in mnemonics:
+            records = self._get_records(mnemonic, starttime, endtime)
+            records.write(cache_path / f'{mnemonic}.ecsv', format='ascii.ecsv')
```

Comment:
Why are we writing out a file and what is it used for?

Reply:
This is the utility needed to cache a set of mnemonics for a time range. These saved files are then used by `EngDB_Mock` to use the service interface offline.